port/win32/omrintrospect.c

```diff
@@ -135,8 +136,7 @@ suspend_all_preemptive(struct PlatformWalkData *data)
 		}
 
 		if (!Thread32First(data->snapshot, &data->thread32)) {
-			result = GetLastError();
-			goto error_return;
```

Comment:
Can you provide a bit more context why you are changing the behaviour here?

Reply:
The code at 'error_return' label does SetLastError(result) and returns -1. So doing GetLastError() and then immediately invoking SetLastError() seems to be a noop. Why don't just return -1 having last error set by Thread32First() function?
See also the code above: CreateToolhelp32Snapshot() also sets last error if it fails, and the check against invalid handle just returns -1 - the last error is set.